### PR TITLE
Use `new (nothrow)` everywhere

### PR DESCRIPTION
**Pull request by peter-b**
We jump through various hoops in order to get new to be recognised as
having nothrow behaviour but it is fragile and doesn't work when the
engine is hosted inside another process (if that process uses its own
implementation of new, as it is entitled to do).

Instead of dealing with this, just bite the bullet and add the
(std::nothrow) argument to every invocation of new in the engine and
externals.

Updated/rebased version of #4593

### CODE REVIEW

**Comment by peter-b** [comment]
This needs an update to `CONTRIBUTING.md` to mention that all uses of `new` need to either be `(nothrow)` or specify placement.

File: engine/src/object.h [hunk 1]
```diff
@@ -450,7 +450,8 @@ struct MCPatternInfo
 	MCPatternRef pattern;
 };
 
-class MCObject : public MCDLlist
+class MCObject :
+  public MCDLlist
 {
```
Comment (livecodeali) [inline engine/src/object.h:455]: Some extraneous whitespace changes here, probably not worth reverting but it proves I was looking properly, right ;-)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 85a7444

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 85a744465b3a4e05d90228bc38efb8aa574322eb

**Comment by livecode-vulcan** [comment]
😞 test failure 85a744465b3a4e05d90228bc38efb8aa574322eb
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/326)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/331)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/326)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/326)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/332)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/326)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/326)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 97b61d1

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 97b61d1c0aaa814d9f6d4795992cac7abd0d0734

**Comment by livecode-vulcan** [comment]
😎 test success 97b61d1c0aaa814d9f6d4795992cac7abd0d0734
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/413)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/418)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/413)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/413)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/419)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/413)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/413)

**Pull request merged**